Add option for Azure cross subscription backups

boxcee opened the pull request:
Closes #1646

skriss commented:
Thanks for this @boxcee! I'll review and do some testing this week.

skriss commented on pkg/cloudprovider/azure/object_store.go:
```diff
@@ -146,22 +147,31 @@ func getStorageAccountKey(config map[string]string) (string, error) {
 
-	// 2. we need config["resourceGroup"], config["storageAccount"]
+	// 2. check whether a different subscription ID was set for backups in config["subscriptionId"]
+	var subscriptionId string
```
I'd probably simplify this section to something like:

```go
subscriptionID := envVars[subscriptionIDEnvVar]
if val := config[subscriptionIdConfigKey]; val != "" {
	subscriptionID = val
}
```

boxcee replied:
Sure, thanks. Just digging into the Go stuff.

boxcee commented:
@skriss For testing, is there anything I should document? We have an Azure account and I will test this today.

skriss commented:
@boxcee just some quick notes on exactly what scenario/steps you tested would be great - don't need anything long, just enough to get the idea. thanks!

boxcee commented:
Soooo, I did test this and it does work. **However**, I am not really sure what value it adds as the snapshots cannot be stored in another subscription. That is at least as far as I got.

I wanted to test this by creating a cluster, creating a backup, deleting the cluster, creating another cluster and then restore the backup to that cluster. Turns out, when I deleted the cluster I also deleted all the snapshots. When I then tried to restore, the snapshots could not be found.

jweldonca commented:
hey @boxcee, how do you test this succesfully? 

When I build you're branch (boxcee:allow-cross-subscription-backups) and then run velero install

I get the following error 
`An error occurred: some backup storage locations are invalid: error getting backup store for location "default": rpc error: code = Unknown desc = config has invalid keys [subscriptionId]; valid keys are [resourceGroup storageAccount bucket prefix]
`

This is the command I've ran
`velero install --provider azure --bucket $BLOB_CONTAINER --secret-file ./credentials-velero --backup-location-config resourceGroup=$rg,storageAccount=$storage,subscriptionId=$subId --snapshot-location-config apiTimeout=10000,resourceGroup=$rg,subscriptionId=$subId`

boxcee commented:
@jweldonca I assume this error shows up in the controller?

By default the `Makefile` creates a new container when run with `all-containers`. I changed the repository to point to our company repository and docker pushed the images there.
Then I also modified the deployment (after installing with velero install) to point to the company repo.

Your problem is basically that you have built a new velero binary, but the images being deployed are still the `master` images.

boxcee commented:
@skriss Can confirm it does work cross-subscription now!

1. Created a cluster in subscription A and configured it to create snapshots and backups in subscription B.
2. Used the example `nginx-app` to create a backup.
3. Deleted the cluster in subscription A.
4. Re-created a cluster in subscription A and configured it as before to use subscription B for snapshots and backups.
5. Then I ran the restore command from the example.
6. Success!

skriss commented:
👍 I didn't realize that cross-subscription volume snapshots were even possible - sounds great! I'll take another look at the code shortly.
